cogs/channel.py

```diff
@@ -0,0 +1,144 @@
+
+        try:
+            return await guild.create_category(category)
+        except:
+            print("Issue with ", category, ".  Error: ", sys.exc_info()[0])
+
+    async def getCategory(category_name, context):
+        """
+        Finds a category by same or if none exists creates a new one
+            Parameters:
+                category: the name of the category
+                context: the context of the command
+
+            Returns:
+                the category object
+        """
+        guild = context.guild
+        category = find(lambda category: category.name ==
+                        category_name, guild.categories)
+        # get category by category name
+        if(category == None):
+            category = await guild.create_category(category_name)
+        return category
+
+    async def createChannel(channel_name: str, category_name: str, context):
+        """
+        Given a channel and a category name, creates a channel and assigns it to the category
+            Parameters:
+                channel_name: the name of the channel
+                category_name: the name of the category
+        """
+        guild = context.guild
+
+        category = await ChannelManager.getCategory(category_name, context)
+
+        return await guild.create_text_channel(channel_name, category=category)
+
+    def makeDict(categories, channels):
+        classDict = {}
+        for channel in channels:
+            for category in categories:
+                if (category in channel):
+                    classDict[channel] = category
+        print(classDict)
+        return classDict
+
+    @commands.command(name="channelparse")
+    @has_permissions(administrator=True)
+    async def channelparse(self, context):
+        """
+        [Optional filename argument] This command sets up channels from a csv
+        """
+
+        channelnames = []
+        categories = []
+
+        # parses command to get message content
+        prefix = config["bot_prefix"]
+        startLen = len(prefix) + len("channelparse")
+        filename = context.message.content[startLen:].strip()
+
+        if(re.search("\W", filename)):
+            await context.send("Please input a filename without special characters or extensions")
+            return
+
+        if (not filename.strip()):
+            await context.send("Please specify a file.")
+            return
+        filename = "resources\\"+filename+".csv"
+        print("filename: ", filename)
+
+        # read and parse the csv
+        with open(filename, newline='') as csvfile:
+            csvreader = csv.reader(csvfile, delimiter=',')
+            for row in csvreader:
+
+                # check if those attributes exist
+                if(row[2].strip() and row[3].strip() and row[19].strip()):
+
+                    classtype = row[2]
+                    # first 3 numbers only (not L1,L2,L3)
+                    classnum = row[3][0:3]
+                    prof = row[19]
```

Comment:
Are you able to turn these `row[#]` calls into variables that describe what they are? It's tough to read and understand easily if you don't already know the structure of the CSV